scheduler/workflow.go

```diff
@@ -315,8 +315,10 @@ func (s *schedulerWorkflow) Start(t *task) {
 	// Block until the job has been either run or skipped.
 	errors := t.manager.Work(j).Promise().Await()
 
-	if len(errors) != 0 {
-		t.RecordFailure(j.Errors())
+	if len(errors) != 0 && len(j.Errors()) > 0 {
```

Comment:
Not sure why there is a change in scheduler for this config rpc change. This also seems to be partially reverting a current commit in Master.